src/lavinmq/exchange/exchange.cr

```diff
@@ -89,19 +89,23 @@ module LavinMQ
     end
 
     def match?(frame : AMQP::Frame)
-      type == frame.exchange_type &&
+      delayed = frame.exchange_type == "x-delayed-message"
+      frame_args = frame.arguments.to_h.dup.reject("x-delayed-type").merge({"x-delayed-exchange" => true})
+      type == (delayed ? frame.arguments["x-delayed-type"] : frame.exchange_type) &&
         @durable == frame.durable &&
         @auto_delete == frame.auto_delete &&
         @internal == frame.internal &&
-        @arguments == frame.arguments.to_h
+        @arguments == (delayed ? frame_args : frame.arguments.to_h)
```

Comment:
You can probably just forward frame values to the other match method to get rid of some code duplication:

```suggestion
match?(frame.type, frame.durable, frame.auto_delete, frame.internal, frame.arguments)
```

Reply:
nice!